templates/macros/_macros-quote.jinja

```diff
@@ -0,0 +1,168 @@
+  {% set has_title = title|length > 0 %}
+  {% set has_heading_row = has_title or has_heading_link %}
+  {% set signpost_image_content = caller('signpost_image') %}
+  {% set has_signpost_image = signpost_image_content|trim|length > 0 %}
+  {% set has_citation_source_name = citation_source_name|trim|length > 0 %}
+  {% set has_citation_source_title = citation_source_title|trim|length > 0 %}
+  {% set has_citation = has_citation_source_name or has_citation_source_title %}
+  {% set cta_content = caller('cta') %}
+  {% set has_cta = cta_content|trim|length > 0 %}
+  {% set image_content = caller('image') %}
+  {% set has_image = image_content|trim|length > 0 %}
+
+  {# Validate heading level selection #}
+  {% if quote_heading_level not in [2, 4] %}
+    {% set quote_heading_level = 2 %}
+  {% endif %}
+
+  {%- macro _quote_body() -%}
+    <div class="p-section--shallow">
+      <blockquote class="p-pull-quote u-no-margin--top u-no-padding--left">
+        <p class="p-heading--{{ quote_heading_level }} u-no-padding--top">
+          <i>
+            &#8220{{ quote_text }}&#8221
+          </i>
+        </p>
+      </blockquote>
+    </div>
+  {%- endmacro %}
+
+  {%- macro _citation_block() -%}
+    {%- if has_citation -%}
+      <!--Optional citation block-->
+      {#- Source name and title must be wrapped in grid if they are both present, for responsive view -#}
+      {%- macro _citation_block_source_name() -%}
+        <!--Optional citation source name-->
+        <h5 class="u-no-margin--bottom u-no-padding--top">{{ citation_source_name }}</h5>
+      {%- endmacro -%}
+      {%- macro _citation_block_source_title() -%}
+        <!--Optional citation source title-->
+        <p class="u-text--muted u-no-padding u-no-margin">{{ citation_source_title }}</p>
+      {%- endmacro -%}
+      <div class="p-section--shallow">
+        <hr class="u-hide--large p-rule--muted">
+        {%- if has_citation_source_name and has_citation_source_title %}
+          <!-- When source name and title are both present, wrap them in a 50/50 row to split them on medium -->
+          <div class="row">
+            <div class="col-3 col-medium-3">
+              {{ _citation_block_source_name() }}
+            </div>
+            <div class="col-3 col-medium-3">
+              {{ _citation_block_source_title() }}
+            </div>
+          </div>
+        {%- else -%}
+          {%- if has_citation_source_name %}
+            <!--Optional citation source name-->
+            {{ _citation_block_source_name() }}
+          {%- endif -%}
+          {%- if has_citation_source_title %}
+            <!--Optional citation source title-->
+            {{ _citation_block_source_title() }}
+          {%- endif -%}
+        {%- endif -%}
+      </div>
+    {%- endif -%}
+  {%- endmacro %}
+
+  {% if has_heading_row %}
+    <!--Optional heading -->
+    <div class="p-section--shallow">
+      <hr class="p-rule--highlight">
```

Comment:
Don't make this change yet, as it hasn't been fully reviewed, but just flagging this now:

This rule's markup will likely need to be changed to `<hr class="is-highlighted">` and be accompanied by the new `hr.is-highlighted` [variant class](https://github.com/canonical/vanilla-framework/pull/5275#issuecomment-2296935760) in order to create the proper spacing below the highlighted rule. It it awaiting team review.

Lyubo's back from vacation so this shouldn't be waiting too much longer - just wanted to keep you in the loop :)

Reply:
Thanks for the heads up, I'll keep an eye on the quotes macro PR

Reply:
The [HR is-highlighted change](https://github.com/canonical/vanilla-framework/pull/5241) was merged, but will be released in [Vanilla 4.16.0](https://github.com/canonical/vanilla-framework/releases/tag/untagged-d8d81ac82b2e58f8dab8) which won't release for a few weeks I think. 

However, this isn't a problem for this PR since you aren't using the quote wrapper with a title (which has a highlighted rule at the top). 

If you find that you need to use the quote wrapper with a title (thus needing the highlighted rule with bottom margin), this can be added to styles temporarily:

```scss
hr.is-highlighted {
  @include vf-highlight-bar($colors--theme--text-default);
  &.is-accent {
    @include vf-highlight-bar($color-accent);
  }
}
```